### ISSUE TITLE
Error feedback in Windows

### ISSUE THREAD

**Issue by profbrady** [Calysto/octave_kernel#34]
Using Jupyter on Windows: If `x` has not yet been assigned a value but is entered in a code cell and executed, the kernel just hangs indefinitely instead of giving an error. If the same is done when using the Python kernel an error is displayed. On OS X and Ubuntu I get the error `error: 'x' undefined near line 1 column 1`, which I don't remember working correctly before today. I tested OS X and Ubuntu with other errors and they displayed, but not in Windows.

I tested all 3 OSs with the following code:

```
x=[1:8]
y=[2:8]
x.*y
```

Both OS X and Ubuntu gave the proper error. Windows hung and gave the following in the terminal:

```
[I 17:06:12.380 NotebookApp] Kernel started: 765d170c-926a-4993-a2a3-07737da68d69
[IPKernelApp] ERROR | Exception in message handler:
Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1251, in _send
    return os.write(self.write_fd, s)
OSError: [Errno 22] Invalid argument

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\metakernel\process_metakernel.py", line 77, in do_execute_direct
    stream_handler=stream_handler)
  File "C:\Anaconda3\lib\site-packages\metakernel\replwrap.py", line 138, in run_command
    send_prompt_emit=val != 2)
  File "C:\Anaconda3\lib\site-packages\metakernel\replwrap.py", line 97, in _expect_prompt
    self.sendline(self.prompt_emit_cmd)
  File "C:\Anaconda3\lib\site-packages\metakernel\replwrap.py", line 87, in sendline
    self.child.sendline(line)
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1263, in sendline
    n = n + self.send(self.linesep)
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1242, in send
    return self._send(s)
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1945, in _send
    return super(spawnu, self)._send(s.encode(self.encoding, self.errors))
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1254, in _send
    raise EOF('Attempted write to a closed file descriptor')
metakernel.pexpect.EOF: Attempted write to a closed file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\ipykernel\kernelbase.py", line 212, in dispatch_shell
    handler(stream, idents, msg)
  File "C:\Anaconda3\lib\site-packages\ipykernel\kernelbase.py", line 370, in execute_request
    user_expressions, allow_stdin)
  File "C:\Anaconda3\lib\site-packages\metakernel\_metakernel.py", line 337, in do_execute
    retval = self.do_execute_direct(code)
  File "C:\Anaconda3\lib\site-packages\octave_kernel\kernel.py", line 107, in do_execute_direct
    super(OctaveKernel, self).do_execute_direct(code, self.Print)
  File "C:\Anaconda3\lib\site-packages\metakernel\process_metakernel.py", line 87, in do_execute_direct
    self._start()
  File "C:\Anaconda3\lib\site-packages\metakernel\process_metakernel.py", line 57, in _start
    self.wrapper.child.terminate()
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1332, in terminate
    self.kill(signal.SIGTERM)
  File "C:\Anaconda3\lib\site-packages\metakernel\pexpect.py", line 1504, in kill
    os.kill(self.pid, sig)
PermissionError: [WinError 5] Access is denied
```

**Comment by blink1073** [comment]
Yes, unfortunately that is a limitation of running this on Windows.  Octave presumes it is running in a pty, which Windows lacks.  When it encounters an error, it forcibly closes the pipe, leaving us hanging.

**Comment by profbrady** [comment]
Bummer. I know there were reasons I preferred OS X and Linux. Thanks for the response.

**Comment by blink1073** [comment]
This is fixed in v0.19!  It took a combination of `pexpect` 4.2 and the `--interactive` flag to `octave`.

**Comment by PierUU** [comment]
Well done !!
